qiskit/pulse/pulse_lib/parametric_pulses.py

```diff
@@ -0,0 +1,399 @@
+            raise PulseError("The amplitude norm must be <= 1, "
+                             "found: {}".format(abs(self.amp)))
+        if self.sigma <= 0:
+            raise PulseError("Sigma must be greater than 0.")
+
+    @property
+    def parameters(self) -> Dict[str, Any]:
+        return {"duration": self.duration, "amp": self.amp, "sigma": self.sigma}
+
+    def __repr__(self):
+        return '{}(duration={}, amp={}, sigma={})' \
+               ''.format(self.__class__.__name__, self.duration, self.amp, self.sigma)
+
+
+class GaussianSquare(ParametricPulse):
+    """A square pulse with a Gaussian shaped risefall on either side:
+
+    .. math::
+
+        risefall = (duration - width) / 2
+
+        0 <= x < risefall
+
+        f(x) = amp * exp( -(1/2) * (x - risefall/2)^2 / sigma^2) )
+
+        risefall <= x < risefall + width
+
+        f(x) = amp
+
+        risefall + width <= x < duration
+
+        f(x) = amp * exp( -(1/2) * (x - (risefall + width)/2)^2 / sigma^2) )
+    """
+
+    def __init__(self,
+                 duration: int,
+                 amp: complex,
+                 sigma: float,
+                 width: float,
+                 name: Optional[str] = None):
+        """Initialize the gaussian square pulse.
+
+        Args:
+            duration: Pulse length in terms of the the sampling period `dt`.
+            amp: The amplitude of the Gaussian and of the square pulse.
+            sigma: A measure of how wide or narrow the Gaussian risefall is; see the class
+                   docstring for more details.
+            width: The duration of the embedded square pulse.
+            name: Display name for this pulse envelope.
+        """
+        self._amp = complex(amp)
+        self._sigma = sigma
+        self._width = width
+        super().__init__(duration=duration)
+
+    @property
+    def amp(self) -> complex:
+        """The Gaussian amplitude."""
+        return self._amp
+
+    @property
+    def sigma(self) -> float:
+        """The Gaussian standard deviation of the pulse width."""
+        return self._sigma
+
+    @property
+    def width(self) -> float:
+        """The width of the square portion of the pulse."""
+        return self._width
+
+    def get_sample_pulse(self) -> SamplePulse:
+        return gaussian_square(duration=self.duration, amp=self.amp,
+                               width=self.width, sigma=self.sigma,
+                               zero_ends=False)
+
+    def validate_parameters(self) -> None:
+        if abs(self.amp) > 1.:
+            raise PulseError("The amplitude norm must be <= 1, "
+                             "found: {}".format(abs(self.amp)))
+        if self.sigma <= 0:
+            raise PulseError("Sigma must be greater than 0.")
+        if self.width < 0 or self.width >= self.duration:
+            raise PulseError("The pulse width must be at least 0 and less than its duration.")
+
+    @property
+    def parameters(self) -> Dict[str, Any]:
```

Comment:
It should be able to extract an ordered list of parameters. I worry this is relying on the implementation of the Python dictionary. I would have preferred `parameters` to return a list and `named_parameters` to have returned the dictionary.

Reply:
does `OrderedDict` + `.values()` work?

Reply:
`self.parameters[k] for k in sorted(self.parameters)`